.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,29 @@
+name: golangci-lint
+on:
+  push:
+    paths-ignore:
+      - "website/**"
+      - "proposals/**"
+      - "hack/**"
+      - "design/**"
+      - "changelog/**"
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v1
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.29
+
+          # Optional: golangci-lint command line arguments.
+          args: --timeout 5m
+          
```

Comment:
Are we switching to github actions? or is this an experiment by starting with the linter?

Reply:
Yes. This PR move the test to GitHub action which also will make comments in the code (pr review) when issues are faced. See my POC: https://github.com/dev4devs-com/lintpoc/pull/2